### ISSUE TITLE
word after "alias ssc='sudo systemctl'; ssc " displayed as error

### ISSUE THREAD

**Issue by blueyed** [zsh-users/zsh-syntax-highlighting#552]
I have an `alias ssc='sudo systemctl'`, and zsh-syntax-highlighting will display `stop` in red with `ssc stop`.

`ssc ls` will not display an error, and `alias ssc=systemctl` (without `sudo`) is also fine.

Bisected to https://github.com/zsh-users/zsh-syntax-highlighting/commit/f3410c5862fc6d89ee3661f7d140e570d6740e8c.

**Comment by danielshahaf** [comment]
Currently only the first word of the alias's expansion is used.  Fixing this in full generality is probably non-trivial, because `alias foo='sudo systemctl &&'` is also valid (and furthermore, the expansion of an alias can contain global aliases).  I suppose we could at least hotfix this by allowing the `:regular:` state for the word after an alias, in order to prevent the false positive you're seeing...  Thoughts?

@blueyed Feel free to join IRC too.

**Comment by blueyed** [comment]
I understand that it uses `sudo` and then the second word is meant to be a command, from/via `sudo`'s completion?  Is that correct`
Could the completion system be used to get the type (after using the whole alias expansion)?
For `alias foo='sudo systemctl &&'` it would be a command then.
(Sorry, I do nothing about the internals here)

**Comment by danielshahaf** [comment]
This is closer, but still wrong.

```diff
diff --git a/highlighters/main/main-highlighter.zsh b/highlighters/main/main-highlighter.zsh
index 455c21f..ec0979c 100644
--- a/highlighters/main/main-highlighter.zsh
+++ b/highlighters/main/main-highlighter.zsh
@@ -69,6 +69,7 @@ _zsh_highlight_highlighter_main_predicate()
 
 # Helper to deal with tokens crossing line boundaries.
 _zsh_highlight_main_add_region_highlight() {
+  (( skip_words == 0 )) || return 0
   integer start=$1 end=$2
   shift 2
 
@@ -395,14 +396,19 @@ _zsh_highlight_main_highlighter_highlight_list()
   # Processing buffer
   local proc_buf="$buf"
   local -a args
+  # How many words to skip (these words are the result of alias expansion and are not added to region_highlight)
+  integer skip_words=0
   if [[ $zsyh_user_options[interactivecomments] == on ]]; then
     args=(${(zZ+c+)buf})
   else
     args=(${(z)buf})
   fi
-  for arg in $args; do
+  while (( ${+args[1]} )); do
+    arg=$args[1]
+    shift args
     # Save an unmunged copy of the current word.
     arg_raw="$arg"
+    (( skip_words <= 0 )) || (( --skip_words ))
 
     # Initialize this_word and next_word.
     if (( in_redirection == 0 )); then
@@ -428,6 +434,7 @@ _zsh_highlight_main_highlighter_highlight_list()
     fi
 
     # Compute the new $start_pos and $end_pos, skipping over whitespace in $buf.
+    if (( skip_words == 0 )); then
     start_pos=$end_pos
     if [[ $arg == ';' ]] ; then
       # We're looking for either a semicolon or a newline, whichever comes
@@ -462,6 +469,7 @@ _zsh_highlight_main_highlighter_highlight_list()
       ((start_pos+=offset))
       ((end_pos=$start_pos+${#arg}))
     fi
+    fi
 
     # Compute the new $proc_buf. We advance it
     # (chop off characters from the beginning)
@@ -516,6 +524,8 @@ _zsh_highlight_main_highlighter_highlight_list()
             # Avoid looping forever on alias a=b b=c c=b, but allow alias foo='foo bar'
             [[ $arg == $reply[1] ]] && break
             arg=$reply[1]
+            args[1,0]=( "${(@)reply[2,-1]}" )
+            (( skip_words += $#reply - 1 ))
             if (( $+seen_arg[$arg] )); then
               res=none
               break
```

**Comment by tonylambiris** [comment]
This is happening for me as well -- seemed to have started very recently too (within the past week or so).
![gnome-shell-screenshot-bgn4qz](https://user-images.githubusercontent.com/14058972/47106578-f36d3700-d214-11e8-933d-16255585ca5e.png)

**Comment by phy1729** [comment]
Fixed in cb8c736 #556.
